ws-client/src/client.rs

```diff
@@ -135,9 +134,10 @@ impl WsClient {
 	///
 	/// Fails when the URL is invalid.
 	pub async fn new(remote_addr: impl AsRef<str>, config: WsConfig) -> Result<Self, Error> {
-		let (sender, receiver) = jsonrpc_transport::websocket_connection(remote_addr.as_ref())
-			.await
-			.map_err(|e| Error::TransportError(Box::new(e)))?;
+		let (sender, receiver) =
+			jsonrpc_transport::websocket_connection(remote_addr.as_ref(), config.max_request_body_size)
```

Comment:
Not sure I like this very much, it seems a bit hacky to add a new argument just for max payload size. What about passing the whole `WsConfig` here? Maybe down the road we'll need to use more config options, dunno, receive window size, throttling or rate limiting etc?

Reply:
Indeed, good point. 
I'm also in favor of removing those very coupled methods to use the builder instead.